extensions/kubectl-cli/src/extension.ts

```diff
@@ -0,0 +1,33 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { cli } from '@podman-desktop/api';
+
+export function activate() {
+  cli.createCliTool({
+    markdownDescription:
+      'A command line tool used to run commands against Kubernetes clusters. It does this by authenticating with the Control Plane Node of your cluster and making API calls to do a variety of management actions. If you are just getting started with Kubernetes, prepare to be spending a lot of time with kubectl.\n\nMore information: [kubernetes.io](https://kubernetes.io/docs/reference/kubectl/)',
+    name: 'kubectl',
+    displayName: 'kubectl',
+    images: {
+      icon: 'icon.png',
+    },
+    version: '',
+    path: '',
```

Comment:
there is something missing here no ? like it should be registered only if we found it for now ?
And if we found it we set the version that we found ?

Reply:
I think I missed that one when we pushed API. Those to fields should be optional. It should possible to register without version and location, then podman-desktop would do detection and fill that fields up.

Reply:
no, we only register for now if it has been found

not found : nothing
found: provides the path of the CLI and provides the version

Reply:
once installation would be available, it could provide an optional installation process but it's not there for now